packages/salesforcedx-vscode-apex-debugger/README.md

```diff
@@ -1,3 +1,146 @@
+1. In VS Code, create a launch configuration for the Apex Debugger.
+    1. To open the Debug view, in the VS Code Activity Bar, click the bug icon (hover text: Debug).
+    1. To open your `launch.json` file, click the gear icon (hover text: Open launch.json).
+    1. Within the `"configurations"` array, add a `"Launch Apex Debugger"` configuration:
+        ```
+        "configurations": [
+            {
+                "name": "Launch Apex Debugger",
+                "type": "apex",
+                "request": "launch",
+                "sfdxProject": "${workspaceRoot}"
+            }
+        ]
+        ```
+    1. Save your `launch.json` file.
+
+## Debug Your Code
+After you’ve set up the Apex Debugger, set breakpoints and start a debugging session. Then, debug your code.
+
+To set a breakpoint, open a `.cls` or `.trigger` file and click the column to the left of the line numbers. Active breakpoints are red. Inactive breakpoints are grey. You can see a list of your breakpoints in the Breakpoints panel of the Debug view.
+
+To start a debugging session, from the configuration dropdown menu at the top of the Debug view, select **Launch Apex Debugger**. Then, click the green play icon (hover text: Start Debugging).
+
+While a debugging session is in progress, any synchronous activity that runs a line of code with a breakpoint causes execution to halt at the breakpoint. While execution is paused, you can inspect the call stack and see the current values of your variables. You can also step through your code, using the Debug actions pane that appears at the top of the editor while a debugging session is in progress, and watch those values change. You can debug up to two threads at a time. For more information, see [Debugging](https://code.visualstudio.com/docs/editor/debugging) in the Visual Studio Code Docs.
+
+## Whitelist Users and Request Types
+To filter which requests are debugged, edit your `launch.json` file to set up whitelisting. If you don’t use whitelisting, all events in your org trigger debugging during a debugging session. Whitelist users or request types to focus only on the events that are relevant to the problem you’re debugging.
+
+Add filters to the `"Launch Apex Debugger"` configuration:  
+```
+    "configurations": [
+        {
+            "name": "Launch Apex Debugger",
+            "type": "apex",
+            "request": "launch",
+            "sfdxProject": "${workspaceRoot}"
+            "userIdFilter": [],
+            "requestTypeFilter": [],
+            "entryPointFilter": ""
+        }
+    ]
+```
+
+To auto-complete potential request type values for `"requestTypeFilter"`, press Ctrl+Space.
+
+To filter by entry point, enter a regular expression as the value for `"entryPointFilter"`. For example, to whitelist requests made by the Visualforce page `MyPage`, enter `".*/apex/MyPage.apexp"`.
+
+## Considerations
```

Comment:
The Considerations bulleted list is long and a bit diff to navigate - what about subheads to break it up?